django_informixdb/fields.py

```diff
@@ -8,3 +9,59 @@ def from_db_value(self, value, expression, connection, *ignore):
         if value:
             return value.rstrip()
         return value
+
+
+class CharToBooleanField(models.CharField):
+    """Copied most of this code from the Django BooleanField object:
+    https://docs.djangoproject.com/en/2.1/_modules/django/db/models/fields/#BooleanField
+    """
+    description = "CharField that hides the Y/N implementation and converts to Booleans"
+
+    def __init__(self, *args, **kwargs):
+        kwargs['max_length'] = 1
+        if kwargs.get('null') is None:
+            kwargs['null'] = True
+
+        super().__init__(*args, **kwargs)
+
+    def to_python(self, value):
+        if self.null and value in self.empty_values:
+            return None
+        if value in (True, False):
+            # 1/0 are equal to True/False. bool() converts former to latter.
+            return bool(value)
+        if value in ('t', 'True', '1', 'Y'):
+            return True
+        if value in ('f', 'False', '0', 'N'):
+            return False
```

Comment:
This is literally what the base BooleanField is doing in Django, I just added the 'Y' & 'N' to it. Do you think our char to bool field should be stricter than the base class? Happy to change it just thought I'd give you the context of the change. What do you think @richard-reece @vrani-reece ?

Reply:
Hmm. You made me think :)

This isn't a generic boolean field, it's to handle our use of "Y" / "N" in the database in place of using real boolean fields. My understanding is that other Informix installations also do this because of the strange relationship that Informix has with booleans.

So on reflection, since the purpose of `to_python()` is to turn the value from the DB into a Python type, I think this code should *just* turn "Y" and "N" into booleans, ignoring the other potential string values (especially the ones that are too long for the hardcoded field size, "True" and "False"). If someone has their own variation on "Y" and "N" then I don't think it's unreasonable to require them to subclass and override `to_python()`.

Um, and now that I'm thinking more about it, I realise that I don't believe `get_prep_value()` is correct - it should be emitting "Y" or "N", not a boolean as it does by just calling `to_python()` - the return value of these two methods should be different types, always. One returns a boolean and the other returns a single char string.

Reply:
My original recommendation was to subclass a BooleanField... ... but you'd also have to tweak, then, what the DB layer sees it as. 

I am inclining towards the use of just "Y" / "N" , but then you restrict the max_length, ( = 1 )  and then it cannot be inherited to add the other variants.